services/api.py

```diff
@@ -951,15 +960,17 @@ class SearchViewSet(munigeo_api.GeoModelAPIView, viewsets.ViewSetMixin, generics
     queryset = Unit.objects.all()
 
     def list(self, request, *args, **kwargs):
+        filters = self.request.query_params
+
         # If the incoming language is not specified, go with the default.
-        self.lang_code = request.query_params.get('language', LANGUAGES[0])
+        self.lang_code = filters.get('language', LANGUAGES[0])
         if self.lang_code not in LANGUAGES:
             raise ParseError("Invalid language supplied. Supported languages: %s" %
```

Comment:
I know this was ParseError in the code originally, but maybe it's not appropriate:

    Raised if the request contains malformed data when accessing request.data

Maybe a ValidationError could be better. However, could this be part of the pydantic validation too?